src/api_client.rs

```diff
@@ -231,16 +231,38 @@ impl Client {
     }
 
     pub async fn get_attestations_from_beacon_block(
+        self,
         id: BlockId,
     ) -> Result<Vec<Attestation>, Error> {
-        unimplemented!("")
+        let result: Value<Vec<Attestation>> = self
+            .get(&format!("eth/v1/beacon/blocks/{id}/attestations"))
+            .await?;
+        Ok(result.data)
     }
 
     pub async fn get_attestations_from_pool(
+        self,
         slot: Option<Slot>,
         committee_index: Option<CommitteeIndex>,
     ) -> Result<Vec<Attestation>, Error> {
-        unimplemented!("")
+        let path: String = "eth/v1/beacon/pool/attestations".to_string();
+        let target = self.endpoint.join(&path)?;
```

Comment:
`join` takes `&str` so you can just:
```suggestion
        let path = "eth/v1/beacon/pool/attestations";
        let target = self.endpoint.join(path)?;
```

this is less code and skips the allocation to a `String`